lib/datasource/docker.js

```diff
@@ -22,8 +22,13 @@ function massageRegistry(input) {
   return registry;
 }
 
-function getRepository(pkgName) {
-  return pkgName.includes('/') ? pkgName : `library/${pkgName}`;
+function getRepository(pkgName, registry) {
+  // The implicit prefix only applies to Docker Hub, not other registries
+  if (!registry) {
```

Comment:
What if someone has docker.io/node ? Is that a valid case we should handle?

Reply:
I have never seen someone do that, but docker pull supports it.

```
 $ docker pull docker.io/node
Using default tag: latest
latest: Pulling from library/node
```

Reply:
Pushed a change to support it and corresponding test.